pkg/compose/create.go

```diff
@@ -445,17 +445,19 @@ func getRestartPolicy(service types.ServiceConfig) container.RestartPolicy {
 	return restart
 }
 
-func mapRestartPolicyCondition(condition string) string {
+func mapRestartPolicyCondition(condition string) container.RestartPolicyMode {
 	// map definitions of deploy.restart_policy to engine definitions
 	switch condition {
 	case "none", "no":
-		return "no"
-	case "on-failure", "unless-stopped":
-		return condition
+		return types.RestartPolicyNo
+	case "on-failure":
+		return types.RestartPolicyOnFailure
+	case "unless-stopped":
+		return types.RestartPolicyUnlessStopped
 	case "any", "always":
-		return "always"
+		return types.RestartPolicyAlways
 	default:
-		return condition
+		return types.RestartPolicyNo
```

Comment:
Previously we passed it through as-is, this is now silently coercing everything unknown to no restart. Should probably just do a type conversion and let Engine error out if it's unknown:
```suggestion
		return container.RestartPolicyMode(condition)
```